utils/node_local_test.py

```diff
@@ -3988,6 +3988,79 @@ def test_pydaos_kv(server, conf):
     daos._cleanup()
     log_test(conf, pydaos_log_file.name)
 
+
+def test_pydaos_kv_obj_class(server, conf):
+    """Test the predefined object class works with KV"""
+
+    with tempfile.NamedTemporaryFile(prefix='kv_objclass_pydaos_',
+                                     suffix='.log',
+                                     delete=False) as tmp_file:
+        log_name = tmp_file.name
+        os.environ['D_LOG_FILE'] = log_name
+
+    daos = import_daos(server, conf)
+
+    pool = server.get_test_pool_obj()
+
+    cont = create_cont(conf, pool, ctype="PYTHON", label='pydaos_cont')
+
+    container = daos.DCont(pool.label, cont.label)
+    failed = False
+    # Write kv1 dictionary with OC_S2 object type
+    kv1 = container.dict('object1', {"Monday": "1"}, "OC_S2")
+    if len(kv1) != 1:
+        failed = True
+        print(f'Expected length of kv object is 1 but got {len(kv1)}')
+
+    # Write kv2 dictionary without any object type,
+    # so in this case we have 4 targets so default object type should be S4
+    kv2 = container.dict('object2', {"Monday": "1", "Tuesday": "2"})
+    if len(kv2) != 2:
+        failed = True
+        print(f'Expected length of kv object is 2 but got {len(kv2)}')
+
+    # Run a command to list the objects
+    cmd = ['cont', 'list-objects', pool.label, cont.label]
+    print('list the objects from container')
+    rc = run_daos_cmd(conf, cmd, use_json=True)
+
+    data = rc.json
+    assert data['status'] == 0, rc
+    assert data['error'] is None, rc
+    assert data['response'] is not None, rc
+
+    # Run a command to get the object layout
+    print('query the object layout')
+    actual_obj_layout = []
+    for obj in data['response']:
+        cmd = ['object', 'query', pool.label, cont.label, obj]
+        rc = run_daos_cmd(conf, cmd, use_json=True)
+
+        query_data = rc.json
+        assert query_data['status'] == 0, rc
+        assert query_data['error'] is None, rc
+        assert query_data['response'] is not None, rc
+        actual_obj_layout.append(query_data['response']['class'])
+
+    # Verify the object has the correct layout used during kv dictionary creation.
+    expected_obj_layout = ['S2', 'S4']
+    for obj in expected_obj_layout:
+        if obj not in actual_obj_layout:
+            failed = True
+            print(f'Expected obj {obj} not found in all {actual_obj_layout}')
+
+    if failed:
+        conf.wf.add_test_case('pydaos kv object test', failure='test failed')
+    else:
+        conf.wf.add_test_case('pydaos kv object test')
```

Comment:
There's no stdout from this test but the result is registered at https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-11952/30/testReport/NLT/core/Unit_Tests___NLT_on_EL_8___pydaos_kv_object_test/